### PR TITLE
Fix Generator.Bind argument not working

### PR DESCRIPTION
**Pull request by HowToDoThis**
### Purpose of this PR

* Fix Generator.Bind not reading or parsing arguments
* Print argument errors cleaner

### Testing status

* Local test working successfully.

### Comments
Because this is my first time summiting at here... So anything that I not following the rules or steps please forgive me!
Not very sure that someone will use arguments to do their options on Generator.Bind, and this might not the best way or efficient way to fix this problems, so let us improve together!

### More Info About This PR
Here is a log about using bind.exe commands like -? or -help from current latest upstream git.

```
D:\Code\Github\FixOpenTK\src\Generator.Bind\bin\Debug\netcoreapp31>bind.exe -?
OpenGL binding generator 4.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Using 'all' generator mode.
Use '-mode:all/gl2/gl4/es10/es11/es20/es30/es31' to select a specific mode.
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\Code\Github\FixOpenTK\src\Generator.Bind\bin\Debug\netcoreapp31\src\Generator.Bind\Specifications\GL2\gl.tm'.
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
   at System.IO.StreamReader..ctor(String path)
   at Bind.XmlSpecReader.ReadTypeMap(String file) in D:\Code\Github\FixOpenTK\src\Generator.Bind\XmlSpecReader.cs:line 123
   at Bind.GL2.Generator.Process() in D:\Code\Github\FixOpenTK\src\Generator.Bind\GL2\Generator.cs:line 105
   at Bind.MainClass.Main(String[] arguments) in D:\Code\Github\FixOpenTK\src\Generator.Bind\Main.cs:line 200
```
```
D:\Code\Github\FixOpenTK\src\Generator.Bind\bin\Debug\netcoreapp31>bind.exe -help
OpenGL binding generator 4.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Argument error (System.ArgumentOutOfRangeException: startIndex cannot be larger than length of string. (Parameter 'startIndex')
   at System.String.Substring(Int32 startIndex, Int32 length)
   at System.String.Substring(Int32 startIndex)
   at Bind.MainClass.Main(String[] arguments) in D:\Code\Github\FixOpenTK\src\Generator.Bind\Main.cs:line 60). Please use the '-?' switch for help.
```

### CODE REVIEW

**Comment by HowToDoThis** [comment]
Found out that specifics argument -in:(Path of Input Directory) also not working as well
```
D:\Code\Github\FixOpenTK\src\Generator.Bind\bin\Debug\netcoreapp31>bind.exe -in:..\..\..\Specifications
OpenGL binding generator 4.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Using 'all' generator mode.
Use '-mode:all/gl2/gl4/es10/es11/es20/es30/es31' to select a specific mode.
Reading opengl types.
Reading C# types.
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\Code\Github\FixOpenTK\src\Generator.Bind\bin\Debug\netcoreapp31\src\Generator.Bind\Specifications\Docs\GL'.
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path)
   at Bind.DocProcessor..ctor(IBind generator) in D:\Code\Github\FixOpenTK\src\Generator.Bind\DocProcessor.cs:line 48
   at Bind.GL2.Generator.Process() in D:\Code\Github\FixOpenTK\src\Generator.Bind\GL2\Generator.cs:line 124
   at Bind.MainClass.Main(String[] arguments) in D:\Code\Github\FixOpenTK\src\Generator.Bind\Main.cs:line 200
```

After some deep finding, found out it using default path instead of Settings.InputPath
https://github.com/opentk/opentk/blob/77f711a1937a3fdcf5382bf17c9d033d59af036b/src/Generator.Bind/DocProcessor.cs#L48-L56

File: src/Generator.Bind/Main.cs
Comment (NogginBops) [inline src/Generator.Bind/Main.cs:65]: Not sure why we are just checking to see if the lengths are different, I think we should decide on a separator and check for it explicitly.

Maybe using a regex like this and using the captures: 
```regex
-(\w+)(:(.+))?
```
or just doing a check like:
```cs
if (a.length > match.Value.Length && a[match.Value.Length] == ':')
{
// the rest...
```
I think we should use `>` instead of `!=` for the string lengths.

Comment (HowToDoThis) [reply src/Generator.Bind/Main.cs:65]: How about using Contains in if statement and substring with IndexOf as below demonstration?
```
    if (a.Contains(":"))
        val = a.Substring(a.IndexOf(':') + 1).Trim();
```

Comment (NogginBops) [reply src/Generator.Bind/Main.cs:65]: Sounds reasonable enough. 
A little less verification then the regex idea but I'm fine with it.

**Comment by HowToDoThis** [comment]
At yesterday comment, I tried to find out which argument not working, after some test, we have -input, -output and -class not working right now.

-input parsed successfully, but Processor not reading and thrown DirectoryNotFoundException.
-output also parsed successfully, but overwritten by Generator settings.

I don't know -class works for what, but as my opinion, I don't think we let user change class name, because why you require to change class?

**Comment by NogginBops** [comment]
@HowToDoThis unfortunately the namespaces have changed back to `OpenTK` instead of `OpenToolkit`. So there are a few merge conflicts for this PR. If you could merge (or maybe rebase if you know your git) those changes we can look at getting this in.

**Comment by HowToDoThis** [comment]
I'm sorry that so long not updating, because recently just ended my examination of high school. As your request, I have updated to OpenTK, but I'm using net5.0 I can't test it right now for you, I will upload a test of this PR changes tomorrow.

**Comment by NogginBops** [comment]
No problem, it's your own contribution effort.
We can only be grateful that you spent your time fixing this. :)

It seems like there still are a few merge conflicts in this PR, so before those are fixed we can't really merge this.
It also seems like CI is failing, but I haven't looked at why it's failing so maybe it's something insignificant.

**Comment by HowToDoThis** [comment]
> No problem, it's your own contribution effort.
> 
> We can only be grateful that you spent your time fixing this. 🙂 
> 
> 
> 
> It seems like there still are a few merge conflicts in this PR, so before those are fixed we can't really merge this.
> 
> It also seems like CI is failing, but I haven't looked at why it's failing so maybe it's something insignificant.

Actually it's because some changes are compatible for user using commands/parameters to generate(like output, input, specific gl version), I dunno why GitHub reporting as conflict.

**Comment by HowToDoThis** [comment]
Okay, I have fixed Conflicts (Because I haven't Update To latest Version of master quite long time ago).
Just now run build.cmd to try again, which works perfectly as default scripts, but don't know why travis reporting not found directory, but seems like appveyor are happy on it.

**Comment by HowToDoThis** [comment]
Just now tested out building in Linux platform, which encounters same problem that occurs in travis same problem, seems like can't found this directory, but Windows platform are normal, and quick test is because end path are different, causes linux enviroment met this weird buggy

```
 --- Updating bindings (rewrite) ---
/home/nopeople123456789/fixopentk> "/home/nopeople123456789/.dotnet/dotnet"  run -f netcoreapp31 -p src/Generator.Bind/Generator.Bind.csproj (In: false, Out: false, Err: false)
OpenGL binding generator 4.0.0.0 for OpenTK.
For comments, bugs and suggestions visit http://github.com/opentk/opentk

Using 'all' generator mode.
Use '-mode:all/gl2/gl4/es10/es11/es20/es30/es31' to select a specific mode.
Reading opengl types.
Reading C# types.
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_direct_state_access.xml
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_ES3_1_compatibility.xml
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_robustness.xml
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_direct_state_access.xml
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_ES3_1_compatibility.xml
Reading: src/Generator.Bind/Specifications/GL2/GL/4.5/ARB_robustness.xml
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/nopeople123456789/fixopentk/src/Generator.Bind/Specifications\Docs/GL'.
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path)
   at Bind.DocProcessor..ctor(IBind generator) in /home/nopeople123456789/fixopentk/src/Generator.Bind/DocProcessor.cs:line 48
   at Bind.GL2.Generator.Process() in /home/nopeople123456789/fixopentk/src/Generator.Bind/GL2/Generator.cs:line 125
   at Bind.MainClass.Main(String[] arguments) in /home/nopeople123456789/fixopentk/src/Generator.Bind/Main.cs:line 206
Finished (Success) 'UpdateBindingsRewrite' in 00:00:39.7444226
```

**Comment by HowToDoThis** [comment]
Okay, seems like Travis and Appveyor are happying on it!

**Comment by NogginBops** [comment]
@HowToDoThis if you consider this PR done you can remove the [WIP] tag so that we can merge it

**Review by varon — APPROVED** [review]

**Comment by varon** [comment]
Thank you for the PR! It's always wonderful to see new contributions like this.

**Comment by varon** [comment]
Please `@` me on discord for a blue name too!

**Pull request merged**
